### PR TITLE
Upgrade to Quarkus 2.5.4

### PR DESCRIPTION
**Pull request by sophokles73**
this addresses #3007

### CODE REVIEW

**Comment by calohmn** [comment]
@sophokles73 Could you rebase and recreate the DEPENDENCIES file?
It looks like the new dependencies have been harvested by clearlydefined now.

**Comment by sophokles73** [comment]
@calohmn I have rebased and regenerated the DEPENDENCIES ...

File: bom/pom.xml [hunk 1]
```diff
@@ -50,24 +50,22 @@
     <logback.version>1.2.9</logback.version>
     <micrometer.version>1.7.6</micrometer.version>
     <mongodb-image.name>mongo:4.4</mongodb-image.name>
-    <mongodb-driver.version>4.3.3</mongodb-driver.version>
+    <mongodb-driver.version>4.3.4</mongodb-driver.version>
     <native.image.name>registry.access.redhat.com/ubi8/ubi-minimal:8.4</native.image.name>
-    <netty.version>4.1.68.Final</netty.version>
-    <netty.tcnative.version>2.0.42.Final</netty.tcnative.version>
-    <opentracing.version>0.33.0</opentracing.version>
```
Comment (calohmn) [inline bom/pom.xml:57]: The `opentracing.version` property is still used in the hono-core pom.xml.

File: legal/src/main/resources/legal/NOTICE.md [hunk 2]
```diff
@@ -64,11 +64,11 @@ http://www.apache.org/licenses/LICENSE-2.0.html.
```
Comment (calohmn) [inline legal/src/main/resources/legal/NOTICE.md:64]: When editing this file already, the year in the `Copyright Holders` section could be updated.

File: services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/store/model/JdbcBasedDeviceDto.java [hunk 3]
```diff
@@ -77,11 +81,18 @@ public static JdbcBasedDeviceDto forRead(final String tenantId, final String dev
                 device,
                 recordJson.getBoolean("auto_provisioned"),
                 recordJson.getBoolean("auto_provisioning_notification_sent"),
-                recordJson.getInstant("created"),
-                recordJson.getInstant("updated_on"),
```
Comment (calohmn) [inline services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/store/model/JdbcBasedDeviceDto.java:81]: The `JsonObject#getInstant` method is also used in `MongoDbBasedTenantDao` and `FileBasedRegistrationService` in a similar fashion. Should it be replaced there as well?

File: bom/pom.xml (hunk 1 above)
Comment (sophokles73) [reply bom/pom.xml:57]: right, we still support embedding the packages into the OSGi bundle ...

Comment (sophokles73) [reply bom/pom.xml:57]: we will get rid of that with Hono 2.0.0

File: services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/store/model/JdbcBasedDeviceDto.java (hunk 3 above)
Comment (sophokles73) [reply services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/store/model/JdbcBasedDeviceDto.java:81]: The problem with the JDBC registry has been introduced by https://github.com/vert-x3/vertx-jdbc-client/issues/231 with vert.x 4.2.0. The strings contained in the *created* and *updated_on* properties are not ISO datetime with timezone strings and therefore need to be handled explicitly as UTC timestamps. In the Mongo DB and the file based DB the instants are stored as proper ISO datetime with timezone strings. Thus, no special handling is required there.

**Comment by sophokles73** [comment]
@calohmn I have pushed corresponding changes

**Review by calohmn — APPROVED** [review]
LGTM

**Pull request merged**